promscale/troubleshooting.md

```diff
@@ -0,0 +1,69 @@
+# Troubleshooting Promscale
+This section contains some ideas for troubleshooting common problems experienced
+with Promscale.
+
+<!---
+* Keep this section in alphabetical order
+* Use this format for writing troubleshooting sections:
+ - Cause: What causes the problem?
+ - Consequence: What does the user see when they hit this problem?
+ - Fix/Workaround: What can the user do to fix or work around the problem? Provide a "Resolving" Procedure if required.
+ - Result: When the user applies the fix, what is the result when the same action is applied?
+* Copy this comment at the top of every troubleshooting page
+-->
+
+## Backing up Promscale data gives an `out of shared memory` error
+<!---
+Cause: Promscale creates a table for each metric. pg_dump and pg_restore lock each table when working on it, so they can require a lot of locks in a single transaction, which can quickly grow higher than the maximum number of locks allowed.
+Consequence: If the number of locks required exceeds the number allowed, Promscale gives an `out of shared memory` error, and the action is aborted.
+Fix: Increase the `max_locks_per_transaction` setting in the PostgreSQL configuration file.
+Result: pg_dump and pg_restore can create the required number of locks, and the backup or restore can continue.
+For more information, see the [backup and restore documentation][promscale-backup-restore].
+-->
+
+## Data is occupying too much space
+<!---
+Cause: Promscale keeps metric data in chunks. The most recent chunk is kept uncompressed as a cache for faster querying, and chunks are compressed as they age. If your data is taking up more disk space than expected, compression might not be working correctly.
+Consequence: If compression is not working correctly, older chunks might not be getting compressed, and taking up too much room on disk.
+Fix: To check that compression is working correctly, query the `prom_info.metric` view and make sure that `total_chunks-number_compressed_chunks` is not larger than 2. If compression is not working correctly: Make sure that you have sufficient background workers (i.e., > number of databases + 2) required to do scheduled jobs like compression and retention. If you are using TimescaleDB version less than 2.0.0, make sure that you are running the maintenance cron jobs, and they are returning success.
+Consequence: When compression is working correctly, your data is being compressed as it ages.
+For more information about compression, see the [Promscale compression documentaion][promscale-compression].
+-->
+
+## The PromQL dashboard doesn't show Promscale data
+<!---
+Cause: Backfilling data which has not been ingested into Prometheus.
+Consequence: The PromQL dashboard does not show Promscale data.
+Fix: Enable the `read_recent` option for Promscale (How??)
```

Comment:
The solution to How is configure `read_recent` in Prometheus configuration in `remote_read` section. 

```
remote_read:
  - url: "http://<connector-address>:9201/read"
    read_recent: true
```